lib/rich_text.rb

```diff
@@ -97,6 +99,22 @@ def linkify(text, mode = :urls)
 
     private
 
+    def linkify_users(text)
+      regex = /(?<!\w)@(?:“([^”]+?)”|”([^”]+?)”|‘([^’]+?)’|’([^’]+?)’|"([^"]+?)"|'([^']+?)'|(\w+))/
```

Comment:
Is there some way to match that doesn't involve an unreadable regular expression?

Reply:
There's also a lot of logic in this regular expression that is not tested. It looks like it matches
- `“”`
- `””`
- `‘’`
- `’’`
- `""`
- `''`

Should we have a test that shows that it doesn't linkify something like `@"foo'`? What happens if there are multiple potential matches? e.g. `@“foo@”bar”`